### DISCUSSION TITLE
Meaning of "required"

### DISCUSSION THREAD

**Discussion by hmarty-ea** [bhch/django-jsonform#128] [General]
It seems that the "required" list in an object means that the value cannot be None, instead of making the property "optional". Do you confirm ? Would be nice to be able to make some properties optional.

**Comment by hmarty-ea** [comment]
Just answered my own question, "additionalProperties" is what I want!

**Comment by bhch** [comment]
Yes, currently, all properties declared in the schema must be present in the data also.

This is only checked/enforced at the validation level. The form widget doesn't really care, though.

I think I'll probably make the validation a bit relaxed in the next version. I.e. if the property is not listed in `required`, then it would be considered optional by default.
